VERSIONING.md

```diff
@@ -0,0 +1,119 @@
+  "CAF": "0.17.6",
+  "Apache Arrow": "6.0.1",
+  "jemalloc": "5.2.1",
+  "plugins": {
+    "compaction": "v1.2.0-61-g9131346958",
+    "matcher": "v2.1.4-23-gfb69e84ade",
+  }
+}
+```
+
+## Version Components
+
+VAST's version number consists of three mandatory and two optional parts:
+`v<major>.<minor>.<patch>[-rc<candidate>][-<tag>]`. Every release on the default
+branch has an associated annotated and gpg-signed tag of the same name.
+
+### Major
+
+A major version increment indicates that the release included breaking changes.
+Such breaking changes include required changes to user configuration, removal of
+previously deprecated features, and updates to the minimum versions of required
+third-party dependencies.
+
+VAST's database format is guaranteed to be forward-compatible between
+consecutive major versions.
+
+### Minor
+
+A minor version increment indicates changes to functionality, e.g., the addition
+of new features and configuration, or the deprecation of existing features and
+configuration.
+
+VAST processes can connect to one another if the major and minor version match.
+
+### Patch
+
+A patch version increment indicates backwards-compatible bug fixes.
+
+### Candidate (optional)
+
+For release candidates, the tag begings with an additional `-rc<number>`, e.g.,
+`v1.2.3-rc1` indicates the first release candidate for the release `v1.2.3`.
+
+### Tag (optional)
+
+VAST's build system uses [`git-describe --abbrev=10 --long
+--dirty`][git-describe] for versioning. This includes a count of commits since
+the last release, the exact commit hash of the build abbreviated to 10
+characters, and an optional `-dirty` suffix if the build contained uncommited
+changes.
+
+The tag is omitted if `git-describe` is unavailable at build time, or the build
+runs outside of the VAST Git repository.
+
+[git-describe]: https://git-scm.com/docs/git-describe
+
+## `libvast` API and ABI Versioning
+
+The API and ABI of the C++ library `libvast` are considered unstable, i.e., VAST
+v1.3.4 will function with `libvast` v1.3.4 only, and not with `libvast` v1.3.5.
+
+The shared object number of `libvast` increaes with every release to indicate
+that every release is considered a breaking change to the ABI. It is calculated
+as follows from the VAST version:
+
+```
+SO number = 1'000'000 * major
+          +     1'000 * minor
+          +           * patch
+```
+
+E.g., the SO name for VAST `v1.23.45-67-g89abcdef12` is `libvast.so.1023045`.
+
+## Plugin Versioning
+
+Plugin versions are optional. If unspecified, VAST's build system re-uses VAST's
+version number. Tenzir encourages plugin authors to version their plugins
+separately.
+
+The version components exactly mirror the version components of VAST. The build
+system uses [`git-describe`] with the last commit to 
+
+Note that due to `libvast`'s unstable ABI, dynamic VAST plugin libraries must
+link against the exact same version of `libvast` as the `vast` binary itself.
+
+## Release Cadence
```

Comment:
This should go into RELEASING.md, it's orthogonal to versioning.

Reply:
I think we can merge the two documents. I will add the content to here.

Reply:
Actually the whole content of RELEASING.md is irrelevant with this document. The CI checks are always enforced, and the name now needs to be decided upon based on the changes. Let's have one place less to maintain and have only this paragraph.

Reply:
I still think it's the wrong document to describe the release procedures (here: cadence) in the document that's about versioning. What does @tenzir/backend think?

Reply:
One alternative would be to simply remove this section.

I would still want to delete RELEASING.md, as the document is both outdated and not adding any value.

Reply:
Okay, let's remove it from this document and rethink it in the context of documentation on our releasing workflows/guidelines.

Reply:
Done.